elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/AggregationDataStoreTransaction.java

```diff
@@ -52,11 +56,21 @@ public void createObject(Object entity, RequestScope scope) {
     @Override
     public Iterable<Object> loadObjects(EntityProjection entityProjection, RequestScope scope) {
         Query query = buildQuery(entityProjection, scope);
-        QueryResult result = queryEngine.executeQuery(query);
-        if (entityProjection.getPagination() != null && entityProjection.getPagination().returnPageTotals()) {
-            entityProjection.getPagination().setPageTotals(result.getPageTotals());
+        queryResult = queryEngine.executeQuery(query);
+        queryResult.run();
```

Comment:
You don't need to call run() here.  Calling queryResult.get() should invoke queryResult.run()